src/node_process_methods.cc

```diff
@@ -130,6 +130,71 @@ static void CPUUsage(const FunctionCallbackInfo<Value>& args) {
   fields[1] = MICROS_PER_SEC * rusage.ru_stime.tv_sec + rusage.ru_stime.tv_usec;
 }
 
+// ThreadCPUUsage use system dependent implementation to get the current thread
+// cpu times:
+//
+// - On Unix, it uses getrusage(2) with RUSAGE_THREAD.
+// - On macOS, it uses thread_info(2) with THREAD_BASIC_INFO.
+// - On Windows, it uses GetThreadTimes(2).
+//
+// Returns those values as Float64 microseconds in the elements of the array
+// passed to the function.
+static void ThreadCPUUsage(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Nit: we can add a fast api call for this as well.

Reply:
I see. I will see what I can do about it.

Reply:
I looked at the documentation but I'm unsure on how to add Fast API to a function returning a complex data structure. Can you give some hints?

Reply:
This is now outdated I think.